solutions/usaco-416.mdx

```diff
@@ -0,0 +1,86 @@
+---
+id: usaco-416
+source: USACO Silver 2014 March
+title: The Lazy Cow
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
+
+## Implementation
+
+**Time Complexity** $\mathcal{O}(K \cdot N ^ 2)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+Depending on how you implement this problem, Java code may have trouble 
+passing the last testcase even though the algorithm is correct.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class TheLazyCow {
```

Comment:
```suggestion
public class Lazy {
```

Reply:
is this optional?

Reply:
yeah ig

Reply:
I think TheLazyCow might be better because it matches the problem name

Reply:
the problem name is Lazy? i always assume that the problem name was the input file name

Reply:
the problem name is the title of the problem, the input file is just a shorter version of the title for file i/o

Reply:
i mean you don't see sol names like WhyDidTheCowCrossTheRoadIII

Reply:
i think some problem names like "WhyDidTheCowCrossTheRoadIII" are too long, but i think TheLazyCow or LazyCow would be better in this case since it's the problem's name and it's not too long

Reply:
ok yeah